plugin/api.go

```diff
@@ -288,6 +288,13 @@ type API interface {
 	// Minimum server version: 5.2
 	CreateTeamMembers(teamId string, userIds []string, requestorId string) ([]*model.TeamMember, *model.AppError)
 
+	// CreateTeamMembersGracefully creates a team membership for all provided user ids and reports the users that were not added
+	//
+	// @tag Team
+	// @tag User
+	// Minimum server version: 5.18
+	CreateTeamMembersGracefully(teamId string, userIds []string, requestorId string) (*model.TeamMembersWithErrors, *model.AppError)
```

Comment:
I would highly vote for not including this into 5.18, given that we can't do breaking changes for the plugin API

Reply:
This is a new api and not a breaking change @hanzei

Reply:
Feel like this is how the original api should've worked in the first place. :D

`CreateTeamMembersBesteffort`?

Reply:
Yes, this is a new API. But the point is that we can't do breaking changes in the plugin API. If we get this API "wrong" (and given the discussion between Ali and I it's not crystal clear what the API should look like), we can't fix this.

I don't see the nee to rush this into 5.18. What do you think about cutting the API changes from the PR and submitting it independently for 5.20?

Reply:
This has already been delayed to 5.19 by Amy

Reply:
Given that 5.19 is a quality release, I would recommend deferring this change to 5.20